find/finder.go

```diff
@@ -113,9 +113,68 @@ func (f *Finder) findRoot(ctx context.Context, root *list.Element, parts []strin
 	return true
 }
 
+var refTypeMap = map[string]string{
+	"datacenter":  "Datacenter",
+	"datastore":   "Datastore",
+	"domain":      "ClusterComputeResource",
+	"dvportgroup": "DistributedVirtualPortgroup",
+	"dvs":         "DistributedVirtualSwitch",
+	"group":       "Folder",
+	"host":        "HostSystem",
+	"network":     "Network",
+	"resgroup":    "ResourcePool",
+	"vm":          "VirtualMachine",
+}
+
+var prefixTypeMap = map[string]struct{ prefix, kind string }{
+	"domain":   {"s", "ComputeResource"},
+	"group":    {"p", "StoragePod"},
+	"resgroup": {"v", "VirtualApp"},
+}
+
+func referenceFromString(arg string) *types.ManagedObjectReference {
+	if ref := object.ReferenceFromString(arg); ref != nil {
+		return ref
+	}
+
+	id := strings.SplitN(arg, "-", 2)
+	if len(id) != 2 {
+		return nil
+	}
+
+	if kind, ok := refTypeMap[id[0]]; ok {
+		if s, ok := prefixTypeMap[id[0]]; ok {
+			if strings.HasPrefix(id[1], s.prefix) {
+				return &types.ManagedObjectReference{
+					Type:  s.kind,
+					Value: arg,
+				}
+			}
+		}
+
+		return &types.ManagedObjectReference{
+			Type:  kind,
+			Value: arg,
+		}
+	}
+
+	return nil
+}
+
 func (f *Finder) find(ctx context.Context, arg string, s *spec) ([]list.Element, error) {
 	isPath := strings.Contains(arg, "/")
 
+	if !isPath {
+		if ref := referenceFromString(arg); ref != nil {
+			p, err := InventoryPath(ctx, f.client, *ref)
+			if err == nil {
```

Comment:
In case err != nil here, should we care? eg.: maybe the client has some issue? context is timed out? (I don't know which cases can happen here, just wondering)

Reply:
If there was a generate connection/auth/etc issue, we'd most likely hit the same on the next API call. But I'll add an explicit check for `ManagedObjectNotFound` which is when we want to fall through to name based lookup.